moveit_core/robot_state/include/moveit/robot_state/cartesian_interpolator.h

```diff
@@ -144,6 +152,75 @@ class CartesianInterpolator
     double meters;
   };
 
+  /** \brief Compute the sequence of joint values that correspond to a straight Cartesian path for a particular link.
+
+     The Cartesian path to be followed is specified as a \e translation vector to be followed by the robot \e link.
+     This vector is assumed to be specified either in the global reference frame or in the local
+     reference frame of the link.
+     The resulting joint values are stored in the vector \e traj, one by one. The interpolation distance in
+     Cartesian space between consecutive points on the resulting path is specified in the \e MaxEEFStep struct which
+     provides two fields: translation and rotation. If a \e validCallback is specified, this is passed to the internal
+     call to setFromIK(). In case of IK failure, the computation of the path stops and the value returned corresponds to
+     the distance that was achieved and for which corresponding states were added to the path.
+
+     The struct CartesianPrecision specifies the precision to which the path should follow the Cartesian straight line.
+     If the deviation at the mid point of two consecutive waypoints is larger than the specified precision, another waypoint
+     will be inserted at that mid point. The precision is specified separately for translation and rotation.
+     The maximal resolution to consider (in percentage of the total path length) is specified by max_resolution.
+  */
+  static Distance computeCartesianPath(
+      const RobotState* start_state, const JointModelGroup* group, std::vector<std::shared_ptr<RobotState>>& traj,
+      const LinkModel* link, const Eigen::Vector3d& translation, bool global_reference_frame,
+      const MaxEEFStep& max_step, const CartesianPrecision& precision,
+      const GroupStateValidityCallbackFn& validCallback = GroupStateValidityCallbackFn(),
+      const kinematics::KinematicsQueryOptions& options = kinematics::KinematicsQueryOptions(),
+      const kinematics::KinematicsBase::IKCostFn& cost_function = kinematics::KinematicsBase::IKCostFn());
+
+  /** \brief Compute the sequence of joint values that correspond to a straight Cartesian path, for a particular link.
+
+     In contrast to the previous function, the translation vector is specified as a (unit) direction vector and
```

Comment:
Does "(unit)" mean that no matter what vector you pass in, it gets normalized first? Maybe worth saying.

Reply:
No, the direction vector is simply multiplied by the distance to yield the final translation vector.
As this can be directly seen from the inline implementation, I don't see the need for explaining this in a comment.

Reply:
This is one of those "if you know what you're doing, you will pass in a unit vector" situations. But not necessary to change, I guess.

Reply:
If you really know what you are doing, it is completely fine to pass a non-unit vector here ;-)
In that case the distance argument, just doesn't determine the travelled distance.

Reply:
Usually, you want to pass a unit vector.